actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -914,15 +914,17 @@ def permitted_scalar?(value)
       #   permitted_scalar_filter(params, "zipcode")
       #
       #   puts params.keys # => ["zipcode"]
-      def permitted_scalar_filter(params, key)
-        if has_key?(key) && permitted_scalar?(self[key])
-          params[key] = self[key]
+      def permitted_scalar_filter(params, permitted_key)
+        if has_key?(permitted_key) && permitted_scalar?(self[permitted_key])
+          params[permitted_key] = self[permitted_key]
         end
 
-        keys.grep(/\A#{Regexp.escape(key)}\(\d+[if]?\)\z/) do |k|
-          if permitted_scalar?(self[k])
-            params[k] = self[k]
-          end
+        permitted_string_key = permitted_key.to_s
```

Comment:
We might not need this value if the block returns on the first line, so perhaps we should inline the `to_s` call.

Reply:
Oh, but then we might end up calling it multiple times. Never mind 🙄